### PR TITLE
1.18.1 support

### PR DESCRIPTION
**Pull request by lewark**
Adds support for Minecraft 1.18.1. Initial tests so far have worked as expected. I've been using this patch for a while with several people using a MC 1.18.1 server and Mumble 1.3.4 and so far it has worked fine.

Changes made in this PR:
- Mod loading events in Forge seem to have been moved exclusively to the specialized mod event bus, so I switched the main MumbleLink mod classes to use this bus.
- IExtensionPoint code has been updated for the newer Forge API.
- Minimum Forge version bumped to 39.0.0, and JVM version bumped to 17

### CODE REVIEW

**Comment by skedgyedgy** [comment]
-- missing file exception when i try to build it myself?
image

**Comment by lewark** [comment]
Odd... I'll give it a look. I didn't bring over the newer Gradle version from my Eclipse build environment so I wonder if that might be the cause.

**Comment by lewark** [comment]
@skedgyedgy I ran into the same problem but the issue should be resolved now. I updated the PR's Gradle version and the mod now builds properly from the command line.

**Review by TVince0 — APPROVED** [review]

File: mod/src/main/java/zsawyer/mods/mumblelink/MumbleLinkImpl.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public void preInit() {
         ModLoadingContext context = ModLoadingContext.get();
         context.registerExtensionPoint(IExtensionPoint.DisplayTest.class
                 , () -> new IExtensionPoint.DisplayTest(
-                        () -> FMLNetworkConstants.IGNORESERVERONLY,
+                        () -> "ANY",
```
Comment (zsawyer) [inline mod/src/main/java/zsawyer/mods/mumblelink/MumbleLinkImpl.java:88]: use a constant

File: mod/src/main/java/zsawyer/mods/mumblelink/addons/pa/es/ExtendedPASupport.java [hunk 2]
```diff
@@ -104,7 +104,7 @@ public void preInit() {
         ModLoadingContext context = ModLoadingContext.get();
         context.registerExtensionPoint(IExtensionPoint.DisplayTest.class
                 , () -> new IExtensionPoint.DisplayTest(
-                        () -> FMLNetworkConstants.IGNORESERVERONLY,
+                        () -> "ANY",
```
Comment (zsawyer) [inline mod/src/main/java/zsawyer/mods/mumblelink/addons/pa/es/ExtendedPASupport.java:107]: use a constant

**Pull request merged**